chutney/engine/src/main/java/com/chutneytesting/engine/domain/execution/engine/step/Step.java

```diff
@@ -415,7 +422,47 @@ private Map<String, Object> getStepOutputs() {
         }
 
         private StepContext copy() {
-            return new StepContext(scenarioContext.unmodifiable(), unmodifiableMap(localContext), unmodifiableMap(evaluatedInputs), unmodifiableMap(stepOutputs));
+            return new StepContext(scenarioContext.unmodifiable(), unmodifiableMap(localContext), unmodifiableMap(evaluatedInputs), unmodifiableMap(stepOutputs), objectMapper).copySnapshotsInputOutput();
+        }
+    }
+
+    public static class StepContextSnapshot {
```

Comment:
Could we have a private inner class here ? as StepContext.

Reply:
Humm ican't because I'm using this class in a test that has to be located in a higher level module (server), do you see another possibility ?